xdp-forward/xdp-forward.c

```diff
@@ -217,6 +222,34 @@ static int do_load(const void *cfg, __unused const char *pin_root_path)
 		pr_info("Loaded on interface %s\n", iface->ifname);
 	}
 
+	/* Attach xdp_check_return to xdp_frame_return tracepoint and
+	 * pin the link to pin_root_path/xdp_check_return.
+	 */
+
+	return_prog = bpf_object__find_program_by_name(skel->obj,"xdp_check_return");
+	if (!return_prog) {
+		pr_warn("Failed to find xdp_check_return program\n");
+		goto end_detach;
+	}
+
+	link = bpf_program__attach_raw_tracepoint(return_prog, "xdp_frame_return");
+	if (libbpf_get_error((const void *) link)) {
+		pr_warn("Couldn't attach to xdp_frame_return: %s\n",
+			strerror(libbpf_get_error((const void *) link)));
+		goto end_detach;
+	}
+	skel->links.xdp_check_return = link;
+
+	snprintf(pin_link_path, sizeof(pin_link_path), "%s/%s", pin_root_path,
```

Comment:
This should use the `try_snprintf()` function (from lib/util), and check the return code. We should probably also disambiguate the pin path right now, with this path bad things will happen if multiple instances of xdp-forward are loaded at the same time.